Sources/Workspace/Workspace.swift

```diff
@@ -3267,6 +3329,68 @@ extension Workspace {
     }
 }
 
+// MARK: - Registry Source archive management
+
+ extension Workspace {
+     func downloadRegistryArchive(
+        package: PackageReference,
+        at version: Version,
+        observabilityScope: ObservabilityScope
+     ) throws -> AbsolutePath {
+         guard let registryManager = self.registryManager else {
+             throw InternalError("registry manager not initialized")
+         }
+
+         let downloadPath = self.location.registryDownloadDirectory.appending(components: package.identity.description, version.description)
+         if !self.fileSystem.exists(downloadPath) {
+             _ = try temp_await {
+                 registryManager.downloadSourceArchive(
```

Comment:
https://github.com/apple/swift-package-manager/pull/3864/files#diff-15980d92e25c769ef0984cd8290b03d5a9bbbf066004db26940d87f845b35781R266

Will need to pass `workspace.checksumAlgorithm`